libcontainer/intelrdt/intelrdt.go

```diff
@@ -241,45 +242,26 @@ func init() {
 
 // Return the mount point path of Intel RDT "resource control" filesysem
 func findIntelRdtMountpointDir() (string, error) {
-	f, err := os.Open("/proc/self/mountinfo")
+	mi, err := mountinfo.GetMounts(func(m *mountinfo.Info) (bool, bool) {
+		// similar to mountinfo.FstypeFilter but stops after the first match
+		if m.Fstype == "resctrl" {
+			return false, true // don't skip, stop
+		}
+		return true, false // skip, keep going
+	})
 	if err != nil {
 		return "", err
 	}
-	defer f.Close()
-
-	s := bufio.NewScanner(f)
-	for s.Scan() {
-		text := s.Text()
-		fields := strings.Split(text, " ")
-		// Safe as mountinfo encodes mountpoints with spaces as \040.
-		index := strings.Index(text, " - ")
-		postSeparatorFields := strings.Fields(text[index+3:])
```

Comment:
I was working with this code because there was a possibility of accessing elements that not exist and that could cause a panic error.
Thanks, @kolyshkin for fixing it and improve code readability.

I adjusted the unit tests to your code. Could you consider using them?

https://github.com/kolyshkin/runc/compare/intel-rdt...Creatone:creatone/intelrdt-moby-sys-mountinfo

Reply:
Left some comments then decided it's easier to rewrite the code. Hope it's OK with you.

Reply:
I'm fine with this :+1: